intranet/apps/dashboard/views.py

```diff
@@ -255,39 +256,36 @@ def get_announcements_list(request, context):
             midnight = timezone.localtime().replace(hour=0, minute=0, second=0, microsecond=0)
             events = Event.objects.visible_to_user(user).filter(time__gte=midnight, show_on_dashboard=True)
 
-    items = sorted(chain(announcements, events), key=lambda item: (item.pinned, item.added))
-    items.reverse()
+    # sort items first by expiration date then by added date
+    items = sorted(chain(announcements, events), key=lambda item: (item.expiration_date, item.added), reverse=True)
```

Comment:
This code completely ignores pinned announcements, for both the main dashboard and archive view. This is not what we intended - the intention was to keep pinned announcements on top for the main dashboard (where expired announcements aren't shown) and only change the sorting order for the archive view (where every announcement is fair game). Sorry if this was unclear.